### ISSUE TITLE
Epic: Refactor Agent Indexing

### ISSUE THREAD

**Issue by avious00** [hyperlane-xyz/hyperlane-monorepo#3281]
## Context
We need to improve the robustness of our agents indexing. Based on this **[Notion document proposal](https://www.notion.so/hyperlanexyz/Indexing-Refactor-7220f23de5fd4592beb30af801b6f61a?pvs=4)** we created this epic to address the top issues.

## Details

**[Notion document](https://www.notion.so/hyperlanexyz/Indexing-Refactor-7220f23de5fd4592beb30af801b6f61a?pvs=4)**

See all issues https://github.com/hyperlane-xyz/hyperlane-monorepo/issues?q=is%3Aissue+is%3Aopen+indexing, need to sift through and decide which are important & plan out what we should do to address them
```[tasklist]
### Tasks
- [ ] https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3224
- [ ] https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3213
- [ ] https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3264
- [ ] https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3267
- [ ] https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3265
- [ ] https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3266
- [ ] https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3268
- [ ] https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3269
```

related: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2313 & other indexing issues https://github.com/hyperlane-xyz/hyperlane-monorepo/issues?q=is%3Aissue+is%3Aopen+indexing

**Comment by avious00** [comment]
hey @tkporter is https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3171 still relevant here? should it be pulled into this epic

**Comment by tkporter** [comment]
thanks for calling it out, closed as it should be fixed now

**Comment by avious00** [comment]
hey @tkporter did this consider https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2308

**Comment by tkporter** [comment]
@avious00 yeah part of https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3264 is to think about this. Let's keep https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2308 open for now
